packages/xdl/src/detach/AndroidShellApp.js

```diff
@@ -1074,6 +1074,55 @@ export async function runShellAppModificationsAsync(context, sdkVersion, buildMo
     );
   }
 
+  // Configure google-services.json
+  if (parseSdkMajorVersion(sdkVersion) >= 37) {
+    // New behavior - googleServicesFile overrides googleSignIn configuration
+    if (!manifest.android || !manifest.android.googleServicesFile) {
+      // No google-services.json file, let's use the placeholder one
+      // and insert keys manually ("the old way").
+
+      // Google sign in
+      await regexFileAsync(
+        /"current_key": "(.*?)"/,
+        `"current_key": "${googleAndroidApiKey}"`,
```

Comment:
Don't we want to check if `googleAndroidApiKey` is defined? I mean to add `googleAndroidApiKey && ( ... )` to the condition. This way we wouldn't need to call `regexFileAsync`.

Reply:
I decided to leave that as it is so that if by any chance we leaked our key in a shell app in the future it wouldn’t end up in someone’s standalone app, but your comment has reminded me of a Turtle feature where people can produce their own shellapps and they could expect the `google-services.json` to be unchanged (if the manifest wouldn’t contain any override).

Given those two different points of view, which one would you consider more _wanted_?

Reply:
After a while I have decided that the possibility of leaking our secrets (which would be shell apps providers' — ours — fault) shouldn't be a reason for adding any particular behavior to XDL. I will change the implementation according to your suggestion.

Reply:
Ok, I see. I'd assumed that the google-services.json is empty (in particular, `current_key` is empty) so if the user hasn't specified `googleAndroidApiKey` we don't need to replace anything. 
And yes, I think we shouldn't add any defense code in xdl for an unlikely event of leaking our secrets.